### ISSUE TITLE
Add in minimum Java version for pack devs

### ISSUE THREAD

**Issue by RyanTheAllmighty** [ATLauncher/ATLauncher#149]
Allow pack devs to specify a java version their pack must run on, examples of correct values below:

```
<java>7</java> <!-- All versions of Java 7 -->
<java>1.7</java> <!-- All versions of Java 7 -->
<java>1.7.0_45</java> <!-- Java 7 version 45 only -->
```

Also maybe add in a minimum version like below:

```
<min-java>7</min-java> <!-- Need any version of Java 7 or later (including later Java versions such as 8) -->
<min-java>1.7</min-java> <!-- Need any version of Java 7 or later (including later Java versions such as 8) -->
<min-java>1.7.0_45</min-java> <!-- Need Java 7 version 45 or later (including later Java versions such as 8) -->
```

And to another extreme, maybe a maximum

```
<max-java>7</max-java> <!-- Only allow Java versions 7 and under (unless min set above) -->
<max-java>1.7</max-java> <!-- Only allow Java versions 7 and under (unless min set above) -->
<max-java>1.7.0_45</max-java> <!-- Only Java 7 version 45 and under (unless min set above) -->
```

Having a set version may not be useful UNLESS we do a Mojang and force a user to use a set version of Minecraft by downloading it and forcibly using that, but this will need to be played with and pack devs input gathered.

**Comment by dmodoomsirius** [comment]
What would happen if someone doesn't have the java version specified?

**Comment by RyanTheAllmighty** [comment]
Most likely take them to the download page for that version.

Having specific java update numbers may not be something good since that's very specific and may cause security issues if users are having to install older versions.

So most likely in saying that, it will be a simple major version option, so you can say the user needs Java 7, or Java 8 or either 7 or 8.

**Comment by Cherry** [comment]
I tend to agree with specific version numbers. It's very rare that a specific version of Java X is required, but more a minimum.

Either way, if this were to be implemented, there needs to be a way for advanced users to proceed regardless, acknowledging that there may be issues. Just because a pack author says their pack runs best on Java 7 shouldn't prevent an advanced user from running it on Java 8 for example.

**Comment by RyanTheAllmighty** [comment]
Oh of course it would be like the ram and permgen warning. If the user doesn't have the right version, it will show a popup when trying to play saying they're using a unsupported version of java or most likely a pack custom message and give the option to open the java download link, MAYBE do auto download and use that like Mojang does (still not sure about this one), or just continue to play.

**Comment by RyanTheAllmighty** [comment]
Fixed with 3.2.7.0

For pack devs, the syntax is explained at https://wiki.atlauncher.com/xml:root#java
